edk2toollib/uefi/edk2/build_objects/dsc.py

```diff
@@ -0,0 +1,486 @@
+# @file dsc.py
+# Data model for the EDK II DSC
+#
+# Copyright (c) Microsoft Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+# There will be some overlap between the objects for DSC files and FDF files
+import collections
+DEFAULT_SECTION_TYPE = "COMMON"
+
+
+class dsc_set(set):
+
+    def __init__(self, allowed_classes=[]):
```

Comment:
The default value of parameter allowed_classes can't be a list or dict. Python creates a persistent object for the list or dictionary. Every subsequent time the function is called, Python uses that same persistent object that was created from the first call to the function.

Reply:
I'll fix this